### DISCUSSION TITLE
Woche 6

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#23] [Student Exchange]
Hallo zusammen,

in dieser Woche stand die Erstellung einer Demo im 🌎 "Hello World"-Stil an.
Dazu haben wir uns entschlossen die geplante 🏠 Grundarchitektur auf einen Texteditor und einen Textviewer anzupassen, um die Komplexität der Demo in einem vernünftigen Rahmen zu halten.
Somit konnten wir den groben Rahmen und das Zusammenspiel der einzelnen Komponenten testen und demonstrieren.
##### Texteditor
Die Umsetzung des 📝 Texteditors haben wir minimalistisch gehalten, da er später in der Form nicht mehr vorkommt und rein zu Demozwecken implementiert ist. Dieser kann soll lediglich dem Textviewer ein 📄 Textdokument zur Verfügung stellen können.

![image](https://github.com/Christian-2003/CrInGE/assets/136824971/3752de0d-040e-4222-9fe1-570de24f4539)

##### Textviewer
Die Darstellung des 📄 Textes mittels Game Engine wurde bereits mittels eines Controllers umgesetzt.
Auch wurde zur Überwachung der 🎮 Userinteraktion bereits der Game Controller benutzt.
![image](https://github.com/Christian-2003/CrInGE/assets/80711177/b48d730d-db4c-40e7-a241-03a8e17d76a4)

##### Arbeitsgruppen
Des weiteren ist unsere 👥 Team jetzt final in Arbeitsgruppen für die Entwicklung und Implementation wie folgt aufgeteilt:
| Arbeitsgruppe | Teammitglieder | Zuständigkeit |
|--------|--------|--------|
| 💾 Persistieren | Kai Nuß | Das Schreiben und lesen von komplexen Dateien wie Karten |
| 📝 Game Editor | Tim Schnur | Umsetzung des Ediors für den Spieleentwicklers zum designen von Karten und Scripts |
| 🎮 Game Engine | Christian Steinbring, Felix Solfronk, Jannik Nunn | Umsetzung der Game Engine mit diversen Grundfunktionen und das Steuern der fertigen Spielelogik |

Da in dieser Woche für die Demo nur minimale Aufgaben zum 💾 persistieren zu erledigen war, wurde das zugehörige Team nicht benötigt.
Auch das 🎮 Game Engine Team wurde aufgrund der reduzierten Komplexität nicht voll eingebunden, weshalb lediglich Christian die Demo Engine implementiert und Jannik Nunn das Starten per exe-Datei übernommen hat.

Damit wäre auch schon das Ende des Blogs für diese Woche erreicht.
Wir freuen uns auch euer Feedback und werden euch nächste Woche weitere Fortschritte berichten.

**Comment by PetrusQn** [comment]
Hi,
Ein paar Fragen haben sich mir beim lesen eures Eintrages gestellt...
Inwiefern wird der eingegebene Text, mittels Game-Engine, dargestellt? Schreibt ihr nicht in ein Textdokument o.Ä. und lest es wieder ein? 
Und wie ist das mit der Überwachung zu verstehen? Was war in euerem Beispiel die Useraktion? Wie wird diese Überwachung umgesetzt? 

Ich bin etwas verwundert, warum Christian den Editor nur für die Demo implementiert hat, obwohl ihr eine Arbeitsgruppe "Game Editor" habt, in der Tim eingetragen ist. 
Aber grundsätzlich finde ich das eine gelungene Demo und bin gespannt wie es weiter geht. 👍

**Reply by TimSchnur** [reply]
Hallo,
erstmal vielen Dank für dein Feedback.

Der Text wird vom Programm eingelesen und dann auf einen Canvas gemalt. Dabei wird Mit einem custom Key listener auf die Pfeiltasten "gehört", ob diese betätigt wurden und anschließend geblättert. Dabei läuft dauerhaft ein Controller, welcher diese Überwachung der Tasten übernimmt.

Außerdem hat Christian lediglich die Game Engine und Tim den Editor umgesetzt.

**Comment by lukpr0** [comment]
Hallo,
ihr schreibt, der Texteditor kommt in der Form später nicht mehr vor. In welcher Form kommt er dann vor? Und warum soll das Spiel einen Text anzeigen können?

**Reply by TimSchnur** [reply]
Hallo Lukas,
kurzgesagt soll das Spiel später keinen Text anzeigen. Allerdings soll dieser Text repräsentativ für eine Karte stehen, da die Implementation der Karten innerhalb der kurzen Zeit so nicht möglich gewesen wäre. Deshalb ist dies nur eine Demo, um einen tatsächlichen grafischen Output zu bekommen.
